### PR TITLE
feat: validate chain id on writing

### PR DESCRIPTION
**Pull request by pikonha**
# Validate chain ID on writing calls

## All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Description

It ensures no writing calls from different chains are allowed on the gateway.

## Related Issue

#135 

## Changes
- [X] New feature implementation
- [ ] Bug fix
- [ ] Code refactoring
- [ ] Documentation update
- [ ] Other (please specify)

### Changes to Core Features:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: packages/client/test/db.e2e.spec.ts [hunk 1]
```diff
@@ -680,6 +711,28 @@ describe('DatabaseResolver', () => {
       expect(addr).to.be.an('null')
     })
 
+    it('should block writing text record with different chain ID', async () => {
```
Comment (alextnetto) [inline packages/client/test/db.e2e.spec.ts:714]: What's the difference between this test and the one in line 412?

File: packages/gateway/cmd/database.ts [hunk 2]
```diff
@@ -50,8 +50,8 @@ const _ = (async () => {
     '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80'
   const rpcURL = process.env.RPC_URL || 'http://localhost:8545'
 
-  const chainID = process.env.CHAIN_ID || '31337'
-  const chain = getChain(parseInt(chainID))
+  const chainID = parseInt(process.env.CHAIN_ID || '31337')
```
Comment (alextnetto) [inline packages/gateway/cmd/database.ts:53]: Good to maintain the same variable name across files. `chainId`

File: packages/gateway/test/api.spec.ts [hunk 3]
```diff
@@ -73,7 +73,7 @@ describe('Gateway API', () => {
     const domains = new Map()
     domains.set(node, domain)
     repo.setDomains(domains)
-    validator = new OwnershipValidator(signatureRecover, [repo])
+    validator = new OwnershipValidator(1, signatureRecover, [repo])
```
Comment (alextnetto) [inline packages/gateway/test/api.spec.ts:76]: What is this 1? Always prefer naming an input through a variable, for code readability.

File: packages/client/test/db.e2e.spec.ts (hunk 1 above)
Comment (pikonha) [reply packages/client/test/db.e2e.spec.ts:714]: one is for subdomains stored on the database and the second one is for 2LD stored on the blockchain

File: packages/gateway/cmd/database.ts (hunk 2 above)
Comment (pikonha) [reply packages/gateway/cmd/database.ts:53]: the difference is that the client is using a flag-based config
image

the other components gather the variables from the .env file.

I'd have to change how things are being initialized to standardize it

File: packages/gateway/test/api.spec.ts (hunk 3 above)
Comment (pikonha) [reply packages/gateway/test/api.spec.ts]: it is the chainId, gonna move it to a variable

**Pull request merged**